### PR TITLE
HDDS-10560. Link rocksdb lib to Ozone rocksdb tools lib relative path instead of absolute path

### PR DESCRIPTION
**Pull request by swamirishi**
## What changes were proposed in this pull request?
Currently the native lib load fails since we hardcode the rocksdbjni lib path while linking with the native lib. We need to link it with a relative path.

The patch looks into setting the RPATH for native lib which will link the rocksdblib corresponding to the location of the native lib.
I am adding the stack overflow link which might be helpful here for reviewing:
https://stackoverflow.com/questions/43330165/how-to-link-a-shared-library-with-cmake-with-relative-path/71321163#71321163

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-10560

## How was this patch tested?
Existing Unit test

### CODE REVIEW

**Comment by adoroszlai** [comment]
Please wait for clean CI run in fork before opening PR.

**Comment by swamirishi** [comment]
Got a clean CI on my fork. Opening the PR

**Comment by swamirishi** [comment]
Screen Shot 2024-03-21 at 03 27 22

File: hadoop-hdds/rocks-native/src/CMakeLists.txt [hunk 1]
```diff
@@ -30,25 +30,33 @@ find_package(JNI REQUIRED)
...
+
 add_library(ozone_rocksdb_tools SHARED ${SOURCE_FILES})
-target_link_libraries(ozone_rocksdb_tools ${linked_libraries})
+
+
+target_link_libraries(ozone_rocksdb_tools PRIVATE ${ROCKSDB_LIB})
+target_link_libraries(ozone_rocksdb_tools PRIVATE rocks_tools)
+set_target_properties(ozone_rocksdb_tools PROPERTIES
+        BUILD_WITH_INSTALL_RPATH FALSE
+        LINK_FLAGS "-Wl,-rpath -Wl,'$ORIGIN'")
```
Comment (smengcl) [inline hadoop-hdds/rocks-native/src/CMakeLists.txt:62]: nit: put a new line at the end of the file

File: hadoop-hdds/managed-rocksdb/src/main/java/org/apache/hadoop/hdds/utils/db/managed/JniLibNamePropertyWriter.java [hunk 2]
```diff
@@ -39,8 +39,8 @@ public static void main(String[] args) {
     String filePath = args[0];
     try (Writer writer = new OutputStreamWriter(
         Files.newOutputStream(Paths.get(filePath)), StandardCharsets.UTF_8)) {
-      writer.write("rocksdbLibName=" +
-          Environment.getJniLibraryFileName("rocksdb"));
+      String libName = Environment.getJniLibraryFileName("rocksdb");
```
Comment (kerneltime) [inline hadoop-hdds/managed-rocksdb/src/main/java/org/apache/hadoop/hdds/utils/db/managed/JniLibNamePropertyWriter.java:42]: Why not use `public static String getRocksDBLibFileName() {`

**Comment by kerneltime** [comment]
The cmake changes look fine. I will try to find some time to give this PR a spin locally and complete the review.

File: hadoop-hdds/rocks-native/src/CMakeLists.txt (hunk 1 above)
Comment (swamirishi) [reply hadoop-hdds/rocks-native/src/CMakeLists.txt]: done

File: hadoop-hdds/managed-rocksdb/src/main/java/org/apache/hadoop/hdds/utils/db/managed/JniLibNamePropertyWriter.java (hunk 2 above)
Comment (swamirishi) [reply hadoop-hdds/managed-rocksdb/src/main/java/org/apache/hadoop/hdds/utils/db/managed/JniLibNamePropertyWriter.java]: done

**Review by kerneltime — APPROVED** [review]

**Comment by swamirishi** [comment]
thank you for the review @kerneltime @smengcl

**Pull request merged**